libfoundation/src/foundation-core.cpp

```diff
@@ -205,9 +205,17 @@ void MCMemoryDeleteArray(void *p_array)
 
 #define HASHFACTOR 2654435761U
 
-hash_t MCHashInteger(integer_t i)
+hash_t MCHashInteger(int64_t i)
```

Comment:
Is this the 'right' thing to do?

'hash_t' can change in size (although there might be an argument that it should remain 32-bit given what hash's tend to be used for)

Wouldn't it be better to have 'sized' hash functions? HashInt32 / HashUInt32 / HashInt64 / HashUInt64